### PR TITLE
Clean up tests to remove errors and warnings

### PR DESCRIPTION
**Pull request by rly**
Just cleaning up while I modify other tests. A few deprecated warnings still remain.

### CODE REVIEW

**Comment by t-b** [comment]
@rly Very nice. Also the commits are pretty neat organized.

File: tests/unit/pynwb_tests/test_base.py [hunk 1]
```diff
@@ -18,21 +18,21 @@ def test_init(self):
     def test_add_container(self):
         ts = TimeSeries('test_ts', [0, 1, 2, 3, 4, 5],
                         'grams', timestamps=[0.0, 0.1, 0.2, 0.3, 0.4, 0.5])
-        self.pm.add_container(ts)
+        self.pm.add_data_interface(ts)
```
Comment (oruebel) [inline tests/unit/pynwb_tests/test_base.py:21]: Should this test be renamed. The test is called ``test_add_container`` but it now tests ``add_data_interface`` instead. The same applies to the test_get_container case below.

**Comment by rly** [comment]
As an aside, at what point do we remove deprecated functionality like `add_container`, `get_container`, and `ClusterWaveforms`?

**Review by oruebel — APPROVED** [review]

**Comment by oruebel** [comment]
> As an aside, at what point do we remove deprecated functionality like `add_container`, `get_container`, and `ClusterWaveforms`?

In terms of versioning, that would be for 1.x rather than a 1.0.x. For now, I would make an issue and assign it to the NWB 2.x milestone.

**Pull request merged**